src/ShipAICmd.cpp

```diff
@@ -1281,24 +1305,60 @@ bool AICmdDock::TimeStepUpdate()
 
 	// state 0,2: Get docking data
 	if (m_state == eDockGetDataStart || m_state == eDockGetDataEnd || m_state == eDockingComplete) {
-		const SpaceStationType *type = m_target->GetStationType();
-		SpaceStationType::positionOrient_t dockpos;
-		type->GetShipApproachWaypoints(port, (m_state == 0) ? 1 : 2, dockpos);
-		if (m_state != eDockGetDataEnd) {
-			m_dockpos = dockpos.pos;
-		}
+		// we are already near the station, what could go wrong?
+		// set the fuel reserve to 0, since the fuel could become lower than the
+		// current reserve during the flight, and the ship will not fly anywhere
+		m_prop->SetFuelReserve(0.0);
 
-		m_dockdir = dockpos.zaxis.Normalized();
-		m_dockupdir = dockpos.yaxis.Normalized(); // don't trust these enough
-		if (type->IsOrbitalStation()) {
-			m_dockupdir = -m_dockupdir;
+		const SpaceStationType *type = m_target->GetStationType();
+		// calculate which we are in the queue.
+		int queue_pos = 0;
+		// since ground stations have a different path for each pad, this queue is
+		// not needed there
+		if (type->IsOrbitalStation())
```

Comment:
No single-line if/for statements please, always use brackets especially with something like this.

Reply:
We have a decent amount of single-body-line if statements without brackets, though I agree in this case that if the body of the if statement is greater than one line it should have brackets.